### ISSUE TITLE
Query Loop: Remove placeholder state

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#48013]
The current Query Loop placeholder breaks the wysiwyg experience in the zoomed out Site View:

<img width="1383" alt="Screenshot 2023-02-13 at 10 48 17" src="https://user-images.githubusercontent.com/846565/218438043-a5f66420-a450-4e97-a620-20461c522d04.png">

Since the Query Loop block now has a "Replace" action in the toolbar which allows you to choose a pattern, we can probably eliminate the current placeholder altogether. Instead of asking the user to choose the initial pattern themselves, we can simply insert a core query pattern and allow them to replace manually if they want/need to.

<details><summary>Original issue</summary>Here's how the Query Loop block appears when empty:

<img width="703" alt="Screenshot 2023-02-13 at 10 46 02" src="https://user-images.githubusercontent.com/846565/218437593-8e0a7d28-f996-4b24-a6e7-fac054518a0d.png">

It works well when you're zoomed in editing a template, but less well when you're browsing in the Site Editor:

<img width="1383" alt="Screenshot 2023-02-13 at 10 48 17" src="https://user-images.githubusercontent.com/846565/218438043-a5f66420-a450-4e97-a620-20461c522d04.png">

There are two issues here:

1. The block looks interactive but it isn't until you engage edit mode
2. It looks nothing like what you see on the frontend which kind of breaks the wysiwyg convention inherent to the frame

To improve things here we can potentially find inspiration in the work we did with Image blocks a little while ago. When the block is not selected you see a more visual preview, and the interactive state is only revealed when the block is selected:

![image](https://user-images.githubusercontent.com/846565/218439499-cf92737c-1589-4aa0-bc0f-39dedc070b53.gif)

If we can do something like this with the Query Loop block, it will work much better in browse view.

cc @WordPress/gutenberg-design </details>

**Comment by paaljoachim** [comment]
That sounds like a very good idea James!

**Comment by jasmussen** [comment]
I think it makes a ton of sense. But it seems like we also might want to think through the placeholder strategy a little bit, so we have a system that we can roll out across text and video in addition to images as it is now. The question we have to ask is: does this gray state need more indication of what it's meant to placehold? Or is it fine as is?

Personally I'd lean towards it being okay to try and take this a bit further as-is, and then revisit. But it seems good to just chat through the options, maybe visualize a bit. I think we tried an icon version at one point?

**Comment by jameskoster** [comment]
Yes good points. There are a couple of things we might try here imo. 

One would be to have a 'visual' placeholder state for all blocks, similar to what we've already created for images. We can then assemble a composition of these to act as the de facto Query placeholder.

It may be something as simple as changing the font to something like [Flow Block](https://fonts.google.com/specimen/Flow+Block):

<img width="511" alt="Screenshot 2023-02-27 at 12 16 05" src="https://user-images.githubusercontent.com/846565/221561602-b85d1a66-874c-4e2e-8509-5db9200a686f.png">

Blocks like Query Loop could invoke this placeholder state, to create appearances like so upon insertion:

<img width="966" alt="Screenshot 2023-02-27 at 12 14 14" src="https://user-images.githubusercontent.com/846565/221561349-efd72286-242f-4d42-8822-99385ca939dd.png">

Then the current placeholder would exposed when you select the Query Loop. Just like how the Image block works.

---

Another option would be to eliminate the Query Loop placeholder altogether and ask the user to select a pattern immediately after inserting the block, similar to what we do when creating a new template:

<img width="907" alt="Screenshot 2023-02-27 at 12 19 10" src="https://user-images.githubusercontent.com/846565/221562150-f6e889b2-3b0e-4752-90da-be4ccb0fd332.png">

If they elect to start blank, then the Query Loop would appear that way – an entirely empty container.

**Comment by jasmussen** [comment]
Nice. In general I think the generic text placeholders have worked well where we use them (i.e. post content) but they probably wouldn't work here since nothing is actually output until you choose a pattern. Unlss

In that light, the abstracted placeholder could be interesting. I wonder if there's a way to take that further. But it would probably have to be the right design — something that distinctly looks like a placeholder, even if it looks differently. I do worry about it missing a boundary 🤔

To take it in a different direction, what if the query loop _always_ output an actual query loop. The most basic one you could have, and on-select you'd be prompted to select from a range of patterns. Not sure if that would be better or worse.

**Comment by jameskoster** [comment]
There might be something to that last idea. Especially if we're able to utilise wp:pattern (https://github.com/WordPress/gutenberg/issues/48458). That would make it easy to shuffle through different Query patterns.

**Comment by jameskoster** [comment]
Tentatively updated the OP based on the latest discussion in this issue.
